Source/WebCore/css/ComputedStyleExtractor.cpp

```diff
@@ -2293,10 +2298,10 @@ static bool isLayoutDependent(CSSPropertyID propertyID, const RenderStyle* style
         if (!renderer || !renderer->isBox())
             return false;
         return !(style && style->marginTop().isFixed() && style->marginRight().isFixed()
-            && style->marginBottom().isFixed() && style->marginLeft().isFixed());
+            && style->marginBottom().isFixed() && style->marginLeft().isFixed()) || !renderer->containingBlock()->style().marginTrim().isEmpty();
```

Comment:
any reason for not using containingBlockHasMarginTrim here? (with changing the interface to std::optional< marginTrimType>)

Reply:
Nope! Changed.